### DISCUSSION TITLE
Add possibility to get "$SETTINGS_PROJECT_URL" variable inside a "button-link" component, to create custom links to the configured website

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#12585] [Feature Requests]
I need to get the "Project Url" that i configured in the settings of my website.

I use it to create "Show Preview" buttons inside some of my collections. Those buttons shall open the website with an id attached to them. etc.

I know there are variables like `$CURRENT_USER.id` that can be used in roles for examples, even though I don't find anything in the docs about them.

But there currently is no way to do something like `$SETTINGS_PROJECT_URL` within the `link` field of a `button-link` element.

More information is in the discussion below :)

**Comment by rijkvanzanten** [comment]
Is this for app-extensions or api-extensions?

**Reply by LasseRosenow** [reply]
Neither.

It's just a simple directus button. That I want to use to open a url:


<table>
<tr><th>It looks like this when you create a new item:</th><th>And it is configured like so:</th></tr>
<tr>
<td>

![Screenshot_20220405-173141_Chrome.png](https://user-images.githubusercontent.com/10547444/161791205-b39ed969-e962-403d-adfe-b474de415a2a.png)

</td>
<td>

![Screenshot_20220405-173228_Chrome.png](https://user-images.githubusercontent.com/10547444/161790936-78bf7a07-a81c-4c22-95ba-c255839c4aea.png)

</td>
</tr>
</table>
But I don't want to type the whole url there, but rather would like to just reuse the url I already configured in the settings page.

**Reply by LasseRosenow** [reply]
The issue is that whenever I change the project url, I would not only have to change it in the settings but also in all my preview buttons.

And I will have to change the project url once this page gets deployed to production.

**Reply by rijkvanzanten** [reply]
Ah gotcha. The button link interface doesn't currently have access to the app's settingsStore values, so there's no way for it to dynamically pull that information in as of right now

**Reply by LasseRosenow** [reply]
Thanks for the clarification. Would be nice to have this feature in the future, so I changed this to a `Feature request`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
